src/wp-admin/includes/class-file-upload-upgrader.php

```diff
@@ -79,7 +79,7 @@ public function __construct( $form, $urlholder ) {
 							self_admin_url( 'plugin-install.php' ),
 							__( 'Return to the Plugin Installer' )
 						);
-						wp_die( __( 'Incompatible Archive.' ) . '<br />' . $plugins_page );
+						wp_die( __( 'Incompatible Archive.' ) . '<br />' . $plugins_page, 422 );
```

Comment:
Thoughts on 415 instead? 

```suggestion
						wp_die( __( 'Incompatible Archive.' ) . '<br />' . $plugins_page, 415 );
```

Reply:
My reasoning for using 422 here over 415 is due to the wp_die call being fired because whilst the file type is correct and processable, the constraints within the code stop the file from being processed due to the content of the file being incorrect:
- **422 - Unprocessable Content**: Indicates that the server understands the content type of the request entity, and the syntax of the request entity is correct, but it was unable to process the contained instructions.
- **415 - Unsupported Media Type**: Indicates that the server refuses to accept the request because the payload format is in an unsupported format.
However I am happy to change this to 415 if this is more accurate?

Reply:
@CallumBW95 I think during our Slack conversation you were planning to change this to 415?

Anyone else have thoughts?